pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/ValidationManager.java

```diff
@@ -264,120 +280,45 @@ public static long computeRealtimeTotalDocumentInSegments(
   }
 
   /**
-   * Computes a list of missing intervals, given a list of existing intervals and the expected frequency of the
-   * intervals.
+   * Computes missing intervals based on the given existing intervals and the expected frequency of the intervals.
    *
-   * @param segmentIntervals The list of existing intervals
-   * @param frequency The expected interval frequency
-   * @return The list of missing intervals
+   * @param intervals List of existing intervals
+   * @param frequency Expected interval frequency
+   * @return List of missing intervals
    */
-  public static List<Interval> computeMissingIntervals(List<Interval> segmentIntervals, Duration frequency) {
-    // Sanity check for frequency
-    if (frequency == null) {
-      return Collections.emptyList();
-    }
-
-    // Default segment granularity to day level if its small than hours.
-    if (frequency.getMillis() < Duration.standardHours(1).getMillis()) {
-      frequency = Duration.standardDays(1);
-    }
+  @VisibleForTesting
+  static List<Interval> computeMissingIntervals(List<Interval> intervals, Duration frequency) {
```

Comment:
Can we mark `intervals` as `@nonnull` here?

Reply:
I want to go with the convention of only annotating **nullable** because most of the arguments should be nonnull by default.